experimental/PlaneCurve/ProjEllipticCurve.jl

```diff
@@ -457,3 +445,94 @@ function toweierstrass(C::ProjPlaneCurve{S}, P::Oscar.Geometry.ProjSpcElem{S}) w
    a6 = u^2*evaluate(G[2], [R(0), X[2], X[3]])
    return T(X[2]^2*X[3] + a1*X[1]*X[2]*X[3] + a3*X[2]*X[3]^2 - X[1]^3 - a2*X[1]^2*X[3] - a4*X[1]*X[3]^2 - a6*X[3]^3)
 end
+
+
+################################################################################
+
+function proj_space(E::ProjEllipticCurve)
+   return base_point(E).parent
+end
+
+################################################################################
+#
+# Interface calling the functions in Hecke
+#
+################################################################################
+
+@doc Markdown.doc"""
+    order(E::ProjEllipticCurve)
+Given an elliptic curve `E` over a finite field $\mathbf F$, computes
+$\#E(\mathbf F)$.
+"""
+function Oscar.order(E::ProjEllipticCurve)
+   return Hecke.order(E.Hecke_ec)
+end
+
+################################################################################
+
+@doc Markdown.doc"""
+    rand(E::ProjEllipticCurve)
+
+Return a random point on the elliptic curve `E` defined over a finite field.
+"""
+function Oscar.rand(E::ProjEllipticCurve)
+   PP = proj_space(E)
+   H = E.Hecke_ec
+   if !Hecke.isshort(H)
+      L = Hecke.short_weierstrass_model(H)
```

Comment:
Isn't it rather expensive to call `short_weierstrass_model` again and again if one wants to e.g. sample 100 different points on the curve?

Reply:
Good remark, I added a function which enables to compute a list of random points.